File: sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/KotlinLanguage.java
/*
 * SonarSource SLang
 * Copyright (C) 2018-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package org.sonarsource.kotlin.plugin;

import org.sonar.api.config.Configuration;
import org.sonar.api.resources.AbstractLanguage;

public class KotlinLanguage extends AbstractLanguage {

  private Configuration configuration;

  public KotlinLanguage(Configuration configuration) {
    super(KotlinPlugin.KOTLIN_LANGUAGE_KEY, KotlinPlugin.KOTLIN_LANGUAGE_NAME);
    this.configuration = configuration;
  }

  @Override
  public String[] getFileSuffixes() {
    String[] suffixes = configuration.getStringArray(KotlinPlugin.KOTLIN_FILE_SUFFIXES_KEY);
    if (suffixes == null || suffixes.length == 0) {
      suffixes = KotlinPlugin.KOTLIN_FILE_SUFFIXES_DEFAULT_VALUE.split(",");
    }
    return suffixes;
  }

}

File: sonar-kotlin-api/src/main/java/org/sonarsource/kotlin/api/common/KotlinLanguage.kt
/*
 * SonarSource Kotlin
 * Copyright (C) 2018-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the Sonar Source-Available License for more details.
 *
 * You should have received a copy of the Sonar Source-Available License
 * along with this program; if not, see https://sonarsource.com/license/ssal/
 */
package org.sonarsource.kotlin.api.common

import org.sonar.api.config.Configuration
import org.sonar.api.resources.AbstractLanguage

class KotlinLanguage(
    private val configuration: Configuration,
) : AbstractLanguage(KOTLIN_LANGUAGE_KEY, KOTLIN_LANGUAGE_NAME) {
    override fun getFileSuffixes(): Array<String> =
        (configuration.getStringArray(KOTLIN_FILE_SUFFIXES_KEY)).let {
            if (it.isNullOrEmpty()) KOTLIN_FILE_SUFFIXES_DEFAULT_VALUE.split(",").toTypedArray()
            else it
        }
}
